src/hotspot/share/runtime/objectMonitor.cpp

```diff
@@ -1178,9 +1178,9 @@ void ObjectMonitor::exit(bool not_suspended, JavaThread* current) {
   _Responsible = NULL;
 
 #if INCLUDE_JFR
-  // get the owner's thread id for the MonitorEnter event
-  // if it is enabled and the thread isn't suspended
-  if (not_suspended && EventJavaMonitorEnter::is_enabled()) {
+  // set _previous_owner_tid for the MonitorEnter event if it is enabled and
+  // the thread isn't releasing the monitor from inside enter()
```

Comment:
nit - need a period and the end of the comment.

Reply:
This is rather a bit too obscure so I think the comment needs to be expanded.
// Set _previous_owner_tid for the MonitorEnter event if it is enabled and
// we legitimately owned this monitor. We can also get here if we need to self-suspend
// in enter(), in which case we never really owned this monitor and so should not record
// our thread id. In that case current_pending_monitor() is non-NULL.